lib/ace/mode/vbscript.js

```diff
@@ -43,17 +43,160 @@ define(function(require, exports, module) {
+                if (val in indentKeywords) {
+                    switch (val) {
+                        case "property":
+                        case "sub":
+                        case "function":
+                            modifiers = "(?:(?:Private|Public(?:\\s+Default)?)\\s+)?";
+                        case "select":
+                        case "do":
+                        case "for":
+                        case "class":
+                        case "while":
+                        case "with":
+                            var checkToken = new RegExp("^\\s*" + modifiers + val,"i");
+                            if (checkToken.test(line)) {
+                                level += indentKeywords[val];
+                            }
+                            break;
+                        case "if":
+                            var singleLineCondition = /^\s*(If)\s+(.)*\s+Then\s+(\S)+/i.test(line);
+                            if (!singleLineCondition)
+                                level += indentKeywords[val];
+                            break;
+                        default:
+                            level += indentKeywords[val];
+                            break;
+                    }
+                }
+            }
+        }
+        // Limit the level to +/- 1 since usually users only indent one level
+        // at a time regardless of the logical nesting level
+        if (level < 0) {
+            return -1;
+        } else if (level > 0) {
+            return 1;
+        } else {
+            return 0;
+        }
+    }
+
+    this.getNextLineIndent = function(state, line, tab) {
+        var indent = this.$getIndent(line);
+        var level = 0;
+
+        var tokenizedLine = this.getTokenizer().getLineTokens(line, state);
+        var tokens = tokenizedLine.tokens;
+
+        if (state == "start") {
+            level = getNetIndentLevel(tokens, line);
+        }
+        if (level > 0) {
+            return indent + tab;
+        } else if (level < 0 && indent.substr(indent.length - tab.length) == tab) {
+            // Don't do a next-line outdent if we're going to do a real outdent of this line
+            if (!this.checkOutdent(state, line, "\n")) {
+                return indent.substr(0, indent.length - tab.length);
+            }
+        }
+        return indent;
+    };
+
+    this.checkOutdent = function(state, line, input) {
+        if (input != "\n" && input != "\r" && input != "\r\n")
+            return false;
+
+        var tokens = this.getTokenizer().getLineTokens(line.trim(), state).tokens;
+
+        if (!tokens || !tokens.length)
+            return false;
+        var val = tokens[0].value.toLowerCase();
+        return ((tokens[0].type == "keyword.control.asp" || tokens[0].type == "storage.type.function.asp") && outdentKeywords.indexOf(val) != -1);
+    };
+
+    this.getMatching = function(session, row, column, tokenRange = true) {
```

Comment:
Could you please convert this to es5 style, not using default arguments, we still do not use es6 to support older browsers.

Reply:
Oh, sure i could